src/core/gui/XournalView.cpp

```diff
@@ -681,28 +685,33 @@ void XournalView::repaintSelection(bool evenWithoutSelection) {
     gtk_widget_queue_draw(this->widget);
 }
 
-void XournalView::setSetsquareView(std::unique_ptr<SetsquareView> setsquareView) {
-    GTK_XOURNAL(this->widget)->setsquareView = std::move(setsquareView);
-}
+auto XournalView::getGeometryToolHandler() -> GeometryToolHandler* { return geometryToolHandler; }
 
-void XournalView::resetSetsquareView() {
-    GTK_XOURNAL(this->widget)->setsquareView.reset();
-    this->control->fireActionSelected(GROUP_SETSQUARE, ACTION_NONE);
-}
+auto XournalView::getGeometryToolController() -> GeometryToolController* { return geometryToolController; }
 
-auto XournalView::getSetsquareView() const -> SetsquareView* {
-    g_return_val_if_fail(this->widget != nullptr, nullptr);
-    g_return_val_if_fail(GTK_IS_XOURNAL(this->widget), nullptr);
+void XournalView::makeGeometryTool(std::string tool) {
```

Comment:
Wouldn't an enum type be better for the parameter here?

Reply:
Yes, I will change that.

Reply:
Done